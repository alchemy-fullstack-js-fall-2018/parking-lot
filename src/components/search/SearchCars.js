import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../routes/routes';


export default function SearchCars({ updateSearchTerm, searchTerm, car }) {


  const carDetails = car.map(car => {
    const { make, model, plate } = car;
    return (
      <tr key={plate}>
        <th>{make}</th>
        <th>{model}</th>
        <th><Link to={ROUTES.CAR_DETAIL.linkTo(plate)}>{plate}</Link></th>
      </tr>
    );
  });

  return (
    <Fragment>
      <label htmlFor="search">Search Cars by Plate</label>
      <input type="text" id="search" name="search" value={searchTerm} onChange={updateSearchTerm}/>
      {car &&
        <table width="100%">
          <tbody>
            <tr>
              <th>Make</th>
              <th>Model</th>
              <th>Plate</th>
            </tr>
            {car && carDetails}
          </tbody>
        </table>
      }



    </Fragment>
  );
}
