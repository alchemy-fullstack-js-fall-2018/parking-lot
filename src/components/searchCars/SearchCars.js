import React, { Fragment, PureComponent } from 'react';
import Cars from '../cars/Cars';
import PropTypes from 'prop-types';

export default class SearchCars extends PureComponent {
  static propTypes = {
    cars: PropTypes.array.isRequired,
    updateSearchTerm: PropTypes.func.isRequired,
    searchTerm: PropTypes.string.isRequired
  };

  render() {
    const { cars, updateSearchTerm, searchTerm } = this.props;

    return (
      <Fragment>
        <form>
          <label htmlFor="searchTerm">Search Cars:  </label>
          <input type="text" value={searchTerm} onChange={updateSearchTerm} />
        </form>

        <Cars cars={cars} />
      </Fragment>
    );
  }
}
