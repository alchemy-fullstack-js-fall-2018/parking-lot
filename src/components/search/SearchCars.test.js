import React from 'react';
import { shallow } from 'enzyme';
import SearchCars from './SearchCars';

describe('SearchCars component', () => {
  const car = [{ make: 'Toyota', model: 'Corolla', plate: 'ABC123' }];

  it('renders a component', () => {
    const wrapper = shallow(<SearchCars car={car}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('changes the search term', () => {
    const updateSearchTerm = jest.fn();
    const wrapper = shallow(<SearchCars car={car} searchTerm={'searcher'} updateSearchTerm={updateSearchTerm} />);
    wrapper.find('#search').simulate('change', { target: { name: 'search', value: 'searched' } });
    expect(updateSearchTerm.mock.calls[0][0].target.value).toEqual('searched');
  });
});
