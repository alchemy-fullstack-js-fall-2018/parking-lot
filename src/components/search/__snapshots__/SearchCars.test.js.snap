// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchCars component renders a component 1`] = `
<Fragment>
  <label
    htmlFor="search"
  >
    Search Cars by Plate
  </label>
  <input
    id="search"
    name="search"
    type="text"
  />
  <table
    width="100%"
  >
    <tbody>
      <tr>
        <th>
          Make
        </th>
        <th>
          Model
        </th>
        <th>
          Plate
        </th>
      </tr>
      <tr
        key="ABC123"
      >
        <th>
          Toyota
        </th>
        <th>
          Corolla
        </th>
        <th>
          <Link
            replace={false}
            to="/cars/ABC123"
          >
            ABC123
          </Link>
        </th>
      </tr>
    </tbody>
  </table>
</Fragment>
`;
